src/source/Ice/Network.c

```diff
@@ -183,21 +183,31 @@ STATUS createSocket(PKvsIpAddress pHostIpAddress, PKvsIpAddress pPeerAddress, KV
 STATUS getIpWithHostName(PCHAR hostname, PKvsIpAddress destIp)
 {
     STATUS retStatus = STATUS_SUCCESS;
+    UINT32 errCode;
+    struct addrinfo *res, *rp;
+    BOOL resolved = FALSE;
 
     CHK(hostname != NULL, STATUS_NULL_ARG);
 
-    struct hostent *entry = gethostbyname(hostname);
-    CHK_ERR(entry != NULL, STATUS_RESOLVE_HOSTNAME_FAILED, "gethostbyname() with errno %s", strerror(errno));
-    CHK_ERR(entry->h_length != 0, STATUS_HOSTNAME_NOT_FOUND, "could not find network address of %s", hostname);
-
-    if (entry->h_addrtype == AF_INET) {
-        destIp->family = KVS_IP_FAMILY_TYPE_IPV4;
-        MEMCPY(destIp->address, entry->h_addr_list[0], IPV4_ADDRESS_LENGTH);
-    } else {
-        destIp->family = KVS_IP_FAMILY_TYPE_IPV6;
-        MEMCPY(destIp->address, entry->h_addr_list[0], IPV6_ADDRESS_LENGTH);
+    CHK_ERR((errCode = getaddrinfo(hostname, NULL, NULL, &res)) == 0, STATUS_RESOLVE_HOSTNAME_FAILED, "getaddrinfo() with errno %s", gai_strerror(errCode));
+
+    for (rp = res; rp != NULL && !resolved; rp = rp->ai_next) {
+        if (rp->ai_family == AF_INET) {
+            struct sockaddr_in *addr = (struct sockaddr_in*)rp->ai_addr;
```

Comment:
Define these at the beginning of the functions. Some compilers will choke on this as you have a jump instruction over the declarations

Reply:
Ah, forgot about that. I'll update it.

Reply:
Updated.